File: GO/BaseballGame_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestBaseballGame(t *testing.T) {
	tables := []struct {
		ops  string
		want int
	}{
		{"5 2 C D +", 30},
		{"5 -2 4 C D 9 + +", 27},
	}

	for _, table := range tables {
		got := BaseballGame(table.ops)

		if got != table.want {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}

File: PHP/BaseballGameTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("BaseballGame.php");

class BaseballGameTest extends TestCase {

    public function testBaseballGame() {
        $game = new BaseballGame();
        //Test 1
        $want = 30;
        $got = $game->calPoints(["5","2","C","D","+"]);
        $this->assertEquals($want, $got);
        //Test 2
        $want = 27;
        $got = $game->calPoints(["5","-2","4","C","D","9","+","+"]);
        $this->assertEquals($want, $got);
    }
}
